### ISSUE TITLE
Invalid prototype (PHP 7.2)

### ISSUE THREAD

**Issue by remicollet** [zendframework/zend-cache#142]
From Fedora QA, with PHP 7.2.0RC5
https://apps.fedoraproject.org/koschei/package/php-zendframework-zend-cache?collection=f28

`PHP Fatal error:  Declaration of Zend\Cache\PatternPluginManager::get($plugin, array $options = Array, $usePeeringServiceManagers = true) must be compatible with Zend\ServiceManager\AbstractPluginManager::get($name, ?array $options = NULL) in /builddir/build/BUILDROOT/php-zendframework-zend-cache-2.7.2-3.fc28.noarch/usr/share/php/Zend/Cache/PatternPluginManager.php on line 23
`

**Comment by webimpress** [comment]
@remicollet I've found it when working on #141 and fixed it in https://github.com/zendframework/zend-cache/pull/141/commits/d967c0d39c48a79c3c37ff84d8658240038f4d78

**Comment by remicollet** [comment]
@webimpress great thanks for the pointer. Indeed, now works as expected

Fedora QA is now ALL green for all zend-* package :)

**Comment by roelvanduijnhoven** [comment]
This issue is still active, and prevents one from using zend-cache on PHP 7.2.

**Comment by C0DE8** [comment]
This issue is still active [**2018-03-22**], and prevents from using **zend-cache** with with "**PatternFactory::factory**" on PHP 7.2.  

**For now my (patch) solution is:**

Zend Framework 3 [**_zendframework/zend-cache   2.7.2_**]
Date: 2018-03-22 (10:10)
File: /vendor/zendframework/zend-cache/src/PatternPluginManager.php
Row: 77

search:
--
    public function get($plugin, array $options = [], $usePeeringServiceManagers = true)
    {


replace
--
    public function get($plugin, array $options = null)
    {
        $usePeeringServiceManagers = true;
        if (3 === func_num_args()) {
            $usePeeringServiceManagers = (bool) func_get_arg(3);
        }


This works for me and do not break the functionallity. Although there are better solutions.  ;-)

@zfbot Please fix this issue.  
@remicollet Please fix this issue.  
 

THX

**Comment by webimpress** [comment]
Fixed in #141, see: https://github.com/zendframework/zend-cache/pull/141/commits/d19e3ac77f2e6f0124b4d72bc51afb903cdaffef

**Comment by C0DE8** [comment]
@webimpress I'm, sorry, but how do I get this fixed version?

`composer update`  

does not update "zendframework/zend-cache" component.  

Even if I set in composer.json (of my app):

`"zendframework/zend-cache": "dev-master",`

I do not get the new version.

**Comment by webimpress** [comment]
@C0DE8 sorry, it's not yet merged yet... I just wanted to say thet the fix is prepared in my PR, but unfortunately it's not released :| You can use my PR if you want, just to test, but I wouldn't recommend on live.

**Comment by C0DE8** [comment]
@webimpress  ah ok :)

I'm on development, so I tried to use your "update" branch:

composer.json

repositories
`    {  
      "type": "vcs",  
      "url": "https://github.com/webimpress/zend-cache.git"  
    },
`
require
`"webimpress/zend-cache": "dev-update",`

Unfortunately I got an error:
![c0de8__github_com__zendframework__zend-cache__issues_142__comopser_error_2018-03-22_1110](https://user-images.githubusercontent.com/20956699/37764287-9e3c52e0-2dc1-11e8-86da-1ad59264b5a3.png)

**Comment by webimpress** [comment]
@C0DE8 use `"zendframework/zend-cache": "dev-update"` in `require` section, and in `repositories` section put:
```
{
    "type": "git",
    "url": "https://github.com/webimpress/zend-cache.git"
},
```

All should work. Let me know if you have any issue.

**Comment by weierophinney** [comment]
Fixed with #148; will release with 2.8.0 by end of week.
